### DISCUSSION TITLE
Create a submit button inside a form, that's disabled attribute depends on other form field states

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#6355] [Q&A]
Hello. I'm using RHF V7 with TypeScript. I'm trying to create a button, that's disabled state depends on the valid and dirty values of other fields inside form (for example Name, Surname, Email). The problem is when i input all the fields without onBlur, the button doesn't change it's state. Please help me to make button that will change it's state according to the values of the fields

`import React from 'react'
import { useForm } from 'react-hook-form'
import { yupResolver } from '@hookform/resolvers/yup'
import { Box, Button, TextField, Typography } from '@material-ui/core'
import * as yup from 'yup'

const schema = yup.object().shape({
    Name: yup.string().min(3).required(),
    Surname: yup.string().min(3).required(),
    Email: yup.string().email().required()
})

type FormInputs = {
    Name: string,
    Surname: string,
    Email: string,
}

export const Form: React.FC = () => {

    const { register, handleSubmit, formState: { errors, isValid, isDirty } } = useForm<FormInputs>({
        resolver: yupResolver(schema),
        mode: 'onTouched',
        reValidateMode: 'onChange',
    })

    const onSubmit = (data: FormInputs) => {
        console.log(data)
    }

    return (
        <>
            <Box border={1} width={1 / 4} textAlign="center" p={3} color="main" borderColor="main">
                <form onSubmit={handleSubmit(onSubmit)}>
                    <Typography variant="h5">
                        Order form
                    </Typography>
                    <Box mb={2}>
                        <TextField
                            label="Name"
                            error={errors.Name?.message ? true : false}
                            helperText={errors.Name?.message}
                            type="text"
                            {...register("Name")}
                        />
                    </Box>
                    <Box mb={2}>
                        <TextField
                            label="Surname"
                            error={errors.Surname?.message ? true : false}
                            helperText={errors.Surname?.message}
                            type="text"
                            {...register("Surname")}
                        />
                    </Box>
                    <Box mb={2}>
                        <TextField
                            label="Email"
                            error={errors.Email?.message ? true : false}
                            helperText={errors.Email?.message}
                            type="text"
                            {...register("Email")}
                        />
                    </Box>
                    <Box mt={2}>
                        <Button variant="contained" disabled={!isValid || !isDirty} color="primary">
                            Secondary
                        </Button>
                    </Box>
                </form>
            </Box>
        </>
    )
}`

**Comment by slugmandrew** [comment]
Hi there,

I think you need to run the validation to get `isValid` or `isDirty` to update their state. Are you expecting the button to be disabled even if the validation hasn't been run yet?

EDIT: The docs say that this behaviour is expected:

![image](https://user-images.githubusercontent.com/153625/130991530-a078983f-8582-411d-b2ca-36966528bfc9.png)


(BTW, you can do multiple line code blocks using three backticks (```) instead of one 🙂)
